core/src/main/scala/org/apache/spark/Dependency.scala

```diff
@@ -148,6 +153,18 @@ class ShuffleDependency[K: ClassTag, V: ClassTag, C: ClassTag](
     }
   }
 
+  def resetShuffleMergeState(): Unit = {
+    _shuffleMergeEnabled = canShuffleMergeBeEnabled()
+    _shuffleMergedFinalized = false
+    mergerLocs = Nil
```

Comment:
Why are we resetting `mergeLocs` ?

Reply:
Since we are anyway going to recompute the shuffle data in the case of indeterminate stage retries, we might as well fetch the latest merger locations right?

Reply:
Particularly in context of colocating mergers for coparents (for joins), it would result in churn, no ?

Reply:
Since all the other subsequent stages are yet to be run or resubmitted again, we would reuse the mergers once it is fetched again right? Also if we are resubmitting due to fetch failure, then we wouldn't know how many nodes (or mergers) lost in that case reusing them can cause more pushed blocks not get merged right? In that case, isn't it better to just get new set of mergers?

Reply:
For child stages, yes - not for sibling stages.

Reply:
Do you feel we should reuse the same mergers? We are resetting only in the case of indeterminate stage and that too because there could have been failures we don't know if those merger locations are still valid or not which is why it is better to get the current active + additional mergers locations. Thoughts?

Reply:
+CC @Victsm who worked on colocating mergers.

Reply:
Since we are only doing this for indeterminate shuffles and for determinate shuffle anyway the retry of a stage will have push disabled. So this is fine for now. Also had a discussion offline with @mridulm on this one. If needed, we can revisit in the future.

Reply:
Sounds good.